sdk/aqueduct/backend/api_client.py

```diff
@@ -337,6 +337,7 @@ def edit_dynamic_engine(
         self,
         action: str,
         integration_id: str,
+        config_delta: DynamicK8sConfig = DynamicK8sConfig(),
```

Comment:
I don't think making a class the default value is good practice in Python because it leads to some potentially unexpected behavior. Better is to use None and initialize the object inside.

The reason is that the class is instantiated once per definition, which means multiple calls will reuse the same class. This can be really confusing if you mutate the config at any point.

Reply:
wow interesting, didn't know that...